libs/ebpfnetsh/programs.cpp

```diff
@@ -306,28 +212,80 @@ handle_ebpf_delete_program(
         return status;
     }
 
-    // TODO: If the program is pinned, unpin the specified program.
-    // TODO (issue #83): call ebpf_program_unpin() once it exists.
+    // TODO(issue #190): If the program is pinned, unpin the specified program.
     // The temporary API ebpf_api_unpin_object requires knowing the name a priori
     // and we have no way to get it yet.
 
     // Remove from our list of programs to release our own reference if we took one.
     // If there are no other references to the program, it will be unloaded.
-    auto found = std::find_if(_programs.begin(), _programs.end(), [filename, section](program_state_t& program) {
-        return program.program_filename == filename && program.program_section == section;
-    });
-    if (found != _programs.end()) {
-        found->clean();
-        _programs.erase(found);
-    } else {
-        std::cout << "Program not found\n";
-        return ERROR_SUPPRESS_OUTPUT;
+    bpf_object* object;
+    bpf_object* next_object;
+    bpf_object__for_each_safe(object, next_object)
```

Comment:
When netsh command is called to delete a program by id, why will `bpf_object` already be present in memory?

Reply:
It will only be if you did "add program" within the same process.  Otherwise it won't be.  Need to add code to cover that case.